cmd/syncthing/gui.go

```diff
@@ -1567,15 +1567,51 @@ func browseFiles(current string, fsType fs.FilesystemType) []string {
 
 	fs := fs.NewFilesystem(fsType, searchDir)
 
-	subdirectories, _ := fs.Glob(searchFile + "*")
+	subdirectories, _ := fs.DirNames("")
 
-	ret := make([]string, 0, len(subdirectories))
+	type match struct {
+		path             string
+		case_differences int
+	}
+
+	matches := make([]match, 0, len(subdirectories))
+
+next_subdir:
 	for _, subdirectory := range subdirectories {
+		var m match
+		for i := 0; i < len(subdirectory) && i < len(searchFile); i += 1 {
+			if subdirectory[i] == searchFile[i] {
+				continue
+			}
+
+			if strings.EqualFold(string(subdirectory[i]), string(searchFile[i])) {
+				m.case_differences += 1
+				continue
+			}
+
+			continue next_subdir
+		}
+
 		info, err := fs.Stat(subdirectory)
 		if err == nil && info.IsDir() {
-			ret = append(ret, filepath.Join(searchDir, subdirectory)+pathSeparator)
+			m.path = filepath.Join(searchDir, subdirectory) + pathSeparator
+			matches = append(matches, m)
 		}
```

Comment:
Why do we even bother doing the path comparison dance if its not a dir?

Reply:
Isn't a stat call costly? I die it this way to avoid unnecessary statt calls

Reply:
Not really, in the context of something triggered essentially manually via a network API call.